File: scratch/search_global_online.php
<?php
// scratch/search_global_online.php

$dir = realpath(__DIR__ . '/../');
$patterns = ['global_online.json', 'global_online.lock'];

echo "Scanning directory: $dir\n";

function scan($path, $patterns) {
    $iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($path));
    foreach ($iterator as $file) {
        if ($file->isDir()) continue;
        $filePath = $file->getPathname();
        
        // Skip vendor, cache, and git directories
        if (strpos($filePath, 'vendor') !== false || 
            strpos($filePath, '.git') !== false || 
            strpos($filePath, 'cache') !== false ||
            strpos($filePath, 'scratch') !== false) {
            continue;
        }
        
        $ext = pathinfo($filePath, PATHINFO_EXTENSION);
        if (in_array($ext, ['php', 'html', 'js'])) {
            $content = file_get_contents($filePath);
            foreach ($patterns as $pattern) {
                if (strpos($content, $pattern) !== false) {
                    echo "Match found for '$pattern' in file: $filePath\n";
                }
            }
        }
    }
}

scan($dir, $patterns);
echo "Scan complete.\n";

File: scratch/search_global_online.py
import os
import re

search_dir = r"d:\Ashik\Shebad 21 may"
patterns = [r"global_online\.json", r"global_online\.lock"]

print("Searching for occurrences of:", patterns)

for root, dirs, files in os.walk(search_dir):
    # Skip vendor and .git directories
    if "vendor" in root or ".git" in root or "cache" in root:
        continue
    for file in files:
        if file.endswith((".php", ".html", ".js")):
            file_path = os.path.join(root, file)
            try:
                with open(file_path, "r", encoding="utf-8", errors="ignore") as f:
                    content = f.read()
                    for pattern in patterns:
                        if re.search(pattern, content):
                            print(f"Match found in: {file_path}")
            except Exception as e:
                print(f"Error reading {file_path}: {e}")
